DEF-2171 Text in game objects

mathiaswking opened the pull request.

mathiaswking committed:
DEF-2171 First editor addition of Label component

mathiaswking committed:
DEF-2171 Removed redundant SetScale message

mathiaswking committed:
DEF-2171 Editor and bob can produce labelc content, and the engine can now load the labelc content

mathiaswking committed:
DEF-2171 Reworked batching code in font_renderer.cpp to work like the sprite batching

Added graphics.max_instances and graphics.max_characters to project settings.

mathiaswking committed:
DEF-2171 Added missing test files

mathiaswking committed:
Review cleanup

RagnarSvenssonKing reviewed (changes requested):
Also make sure there are tests for:
- Editor file reloading (e.g. fonts are reloaded and updates the label)
- Engine hot-reloading

mathiaswking committed:
Review fixes

mathiaswking committed:
DEF-2171 Fixed editor integration test

mathiaswking committed:
DEF-2171 Added integrationtest "builtins" content for label component

mathiaswking committed:
DEF-2171 Added label wizard, and reference code.
Added integration tests for label

mathiaswking committed:
DEF-2171 Fixed proto description: added resource type to font + material
Added refactor test for embedded components

mathiaswking committed:
Unit test fix

mathiaswking committed:
DEF-2171 Added label lua script module
Added more label properties (color, outline, shadow)
Fixed material constants and blend mode
Added more documentation

mathiaswking committed:
Compile fix

mathiaswking committed:
Fixed issue with label editor not updating its font in the view
